### PR TITLE
secure client api

### PR DESCRIPTION
**Pull request by allacherkes**

### CODE REVIEW

File: src/unleash_client_wrapper.ts [hunk 1]
```diff
@@ -1,28 +1,33 @@
...
+    client.once("registered", () => {
       log(`FT is initialized to server for extension ${extensionName}. EndPoint: ${serverArgs.ftServerEndPoint}`);
       resolve();
     });
   });
 }
 
 export async function initializeUnleashClient(extensionName: string, serverArgs: ServerArgs, featureStrategies: Strategy[]): Promise<Unleash> {
+  const authorizationHeader: CustomHeaders = {};
+  authorizationHeader["authorization"] = "client_key";
```
Comment (liorfr) [inline src/unleash_client_wrapper.ts:22]: TODO: "client_key" should be entered externally

File: src/unleash_client_wrapper.ts [hunk 2]
```diff
@@ -1,28 +1,33 @@
...
-    client.once("ready", () => {
+    client.once("registered", () => {
       log(`FT is initialized to server for extension ${extensionName}. EndPoint: ${serverArgs.ftServerEndPoint}`);
       resolve();
     });
   });
 }
 
 export async function initializeUnleashClient(extensionName: string, serverArgs: ServerArgs, featureStrategies: Strategy[]): Promise<Unleash> {
+  const authorizationHeader: CustomHeaders = {};
```
Comment (liorfr) [inline src/unleash_client_wrapper.ts:21]: do it in one line and directly in customHeaders i.e.
`
customHeaders: { authorization: "client_key" }
`

File: src/api.ts [hunk 3]
```diff
@@ -2,23 +2,30 @@ import { Unleash } from "unleash-client";
...
-  //check if the feature is enabled
-  //fallback value is false (3rd parameter)
-  return client.isEnabled(`${extensionName}.${featureToggleName}`, context, false);
+    const context: Context = contextManager.getContext(extensionName);
+
+    //check if the feature is enabled
+    //fallback value is false (3rd parameter)
+    return client.isEnabled(`${extensionName}.${featureToggleName}`, context, false);
+  } catch (e) {
+    log(e);
```
Comment (liorfr) [inline src/api.ts:28]: what is the exception ? is it clear that this is an authentication issue ?
please also add a message saying "due to authentication error feature toggle is disabled"

File: src/unleash_client_wrapper.ts (hunk 1 above)
Comment (allacherkes) [reply src/unleash_client_wrapper.ts]: done

File: src/unleash_client_wrapper.ts (hunk 2 above)
Comment (allacherkes) [reply src/unleash_client_wrapper.ts]: done

File: src/api.ts (hunk 3 above)
Comment (allacherkes) [reply src/api.ts:28]: It catches all exceptions, also errors that where thrown (like missing env variables and etc).
Enables app to run without stopping and dispalying disabled status.

**Comment by lidortal** [comment]
hi @liorfr @i059687 , is this still relevant ? do we already have secure client api in place?

**Comment by allacherkes** [comment]
Not relevant, secure_client_api is provided by appstudio proxy

**Comment by liorfr** [comment]
@lidortal,
secure client api is in place. this is an old PR. I have closed it

**Pull request closed without merging**